backend/src/main/kotlin/services/UserService.kt

```diff
@@ -0,0 +1,18 @@
+package services
+
+import ua.ukma.edu.danki.models.UserAuthRequest
+import ua.ukma.edu.danki.models.UserRegisterRequest
+
+interface UserService {
+    /**
+     * @param userAuthRequest - object with info of the user to be authenticated
+     * @return JWT for the user, or null if unsuccessful
+     */
+    suspend fun authenticateUser(userAuthRequest: UserAuthRequest): String?
+
+    /**
+     * @param userRegisterRequest - object with info of the user to be registered
+     * @return whether user is successfully registered
+     */
+    suspend fun registerUser(userRegisterRequest: UserRegisterRequest): Boolean
```

Comment:
Better to throw an error when you can't register user. If we modify the registration somehow then the code might return more than one type of error.